src/nnue/features/half_kpk.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+    Stockfish, a UCI chess playing engine derived from Glaurung 2.1
+    Copyright (C) 2004-2020 The Stockfish developers (see AUTHORS file)
+
+    Stockfish is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Stockfish is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+//Definition of input features HalfKP of NNUE evaluation function
```

Comment:
Could you please explain what HalfKPK means?  I was confused with HalfKKP (King-King-Piece) when I see this one for the first time.

Reply:
As with PK, I'll change the naming and clarify. I didn't really think the names through and I defy the old convention where KPK would be a product of KPK, whereas in this case it's K * (P | K).